include/modem/lte_lc.h

```diff
@@ -751,40 +751,109 @@ enum lte_lc_modem_evt {
-	 *       - mfw_nrf91x1 >= v2.0.0
+	 *       - mfw_nrf91x1
 	 *       - mfw_nrf9151-ntn
 	 */
-	LTE_LC_MODEM_EVT_CE_LEVEL_0,
+	LTE_LC_MODEM_EVT_RF_CAL_NOT_DONE,
 
 	/**
-	 * Selected CE level in RACH procedure is 1, see 3GPP TS 36.331 for details.
+	 * No suitable cells available.
 	 *
 	 * @note This is only supported by the following modem firmware:
-	 *       - mfw_nrf91x1 >= v2.0.0
 	 *       - mfw_nrf9151-ntn
 	 */
-	LTE_LC_MODEM_EVT_CE_LEVEL_1,
+	LTE_LC_MODEM_EVT_NO_CELL_AVAILABLE,
 
 	/**
-	 * Selected CE level in RACH procedure is 2, see 3GPP TS 36.331 for details.
+	 * Suitable cells available.
 	 *
 	 * @note This is only supported by the following modem firmware:
-	 *       - mfw_nrf91x1 >= v2.0.0
 	 *       - mfw_nrf9151-ntn
 	 */
-	LTE_LC_MODEM_EVT_CE_LEVEL_2,
+	LTE_LC_MODEM_EVT_CELL_AVAILABLE,
 
 	/**
-	 * Selected CE level in RACH procedure is 3, see 3GPP TS 36.331 for details.
+	 * Invalid band configuration.
+	 *
+	 * The associated payload is the @c lte_lc_modem_evt.invalid_band_conf member of type
+	 * @ref lte_lc_invalid_band_conf in the event.
+	 *
+	 * @note This is only supported by the following modem firmware:
+	 *       - mfw_nrf91x1
+	 *       - mfw_nrf9151-ntn
+	 */
+	LTE_LC_MODEM_EVT_INVALID_BAND_CONF,
+
+	/**
+	 * Current country the device is operating in.
+	 *
+	 * The associated payload is the @c lte_lc_modem_evt.detected_country member of type
+	 * `uint32_t` in the event. The country is reported as a Mobile Country Code (MCC).
 	 *
 	 * @note This is only supported by the following modem firmware:
-	 *       - mfw_nrf91x1 >= v2.0.0
 	 *       - mfw_nrf9151-ntn
 	 */
-	LTE_LC_MODEM_EVT_CE_LEVEL_3,
+	LTE_LC_MODEM_EVT_DETECTED_COUNTRY
+};
+
+/** Band configuration status. */
+enum lte_lc_band_conf_status {
+	/** Usable bands available in system. */
+	LTE_LC_BAND_CONF_STATUS_OK = 0,
+
+	/** No usable bands available in system. */
+	LTE_LC_BAND_CONF_STATUS_INVALID = 1,
+
+	/** System support not configured. */
+	LTE_LC_BAND_CONF_STATUS_SYSTEM_NOT_SUPPORTED = 2
+};
+
+/** Detected conflicting band lock or operator restrictions. */
+struct lte_lc_invalid_band_conf {
+	/** LTE-M band configuration status. */
+	enum lte_lc_band_conf_status status_ltem;
+
+	/** NB-IoT band configuration status. */
+	enum lte_lc_band_conf_status status_nbiot;
+
+	/** NTN NB-IoT band configuration status. */
+	enum lte_lc_band_conf_status status_ntn_nbiot;
+};
+
+/** Modem domain event.
+ *
+ * This structure is used as the payload for event @ref LTE_LC_EVT_MODEM_EVENT.
+ */
+struct lte_lc_modem_evt {
+	/** Event type. */
+	enum lte_lc_modem_evt_type type;
+
+	/** Payload for the event (optional). */
+	union {
+		/** Payload for event @ref LTE_LC_MODEM_EVT_CE_LEVEL. */
+		enum lte_lc_ce_level ce_level;
+
+		/** Payload for event @ref LTE_LC_MODEM_EVT_INVALID_BAND_CONF. */
+		struct lte_lc_invalid_band_conf invalid_band_conf;
+
+		/** Payload for event @ref LTE_LC_MODEM_EVT_DETECTED_COUNTRY. */
+		uint32_t detected_country;
```

Comment:
I would mention also here that this is MCC like we do for `LTE_LC_EVT_TAU_PRE_WARNING` and its integer value.